buildchain/buildchain/salt_tree.py

```diff
@@ -153,6 +153,25 @@ def _get_parts(self) -> Iterator[str]:
         return parts
 
 
+def salt_embed_text(path: Path, indent: int) -> str:
+    """Return the content a text file in usable form by Salt."""
+    data = ['|']
+    with path.open(encoding='utf-8') as fp:
+        for line in fp:
+            data.append('{}{}'.format(' '*indent, line.rstrip()))
+    return '\n'.join(data)
+
+
+def salt_embed_bytes(path: Path, indent: int) -> str:
+    """Return the content of a binary file in usable form by Salt."""
+    data = ['|']
+    bytestring = path.read_bytes()
+    b64data = base64.encodebytes(bytestring).decode('utf-8')
+    for line in b64data.split('\n'):
+        data.append('{}{}'.format(' '*indent, line.rstrip()))
+    return '\n'.join(data)
```

Comment:
Mmpf, that's a strange way to do things. Both could rely on `yaml.dump` (see https://github.com/scality/metalk8s/pull/2042/commits/1ada06c5769e987afc7473cb58250419e413b5bd for an example of how to customize representers). But let's keep it for a debt ticket.